### DISCUSSION TITLE
Schema for time-series data

### DISCUSSION THREAD

**Discussion by laurent512** [prisma/prisma#16293] [Q&A] [answered]
Hello everyone, 

I am building a dashboard using nextjs and prisma (+ postgresql) 
The dashboard is user specific and will show plots (e.g. € spent by the user over time) 
It is therefore required to manage (read & modify ) time-serie & user specific data

Do you think prisma & postgresql are suitable candidate techno for that ? 
If so, would you have a schema example to achieve it ? 

Thank you so much

**Comment by nurul3101** [comment, marked as answer]
Hello @laurent512 

PostgreSQL and Prisma should work, though [Timescaledb](https://www.timescale.com/) would be the perfect candidate for your case. 
Timescaledb is an extension to PostgreSQL database. Prisma _*works*_ with it though it is not officially supported by Prisma yet. 
You can have a look at this GitHub Issue: [#3228](https://github.com/prisma/prisma/issues/3228) and this GitHub Discussion: [#6963](https://github.com/prisma/prisma/discussions/6963) to know more about caveats of using Prisma with TimescaleDB.

**Comment by laurent512** [comment]
Thank you for the information. 
And concerning the schema, is the following a suitable pattern ?  

model cost_data {
owner : user_id
timestamp : DateTime
value : Int
}

I would appreciate if you would have some information about that as well :)

**Reply by nurul3101** [reply]
Hello @laurent512 

In the model that you have specified, the owner field looks like a foreign key, so you can add a relation field to the cost_date model as well. Also, to store timestamps, the native @db.timestamp attribute can be used - [reference](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#postgresql-6).

So it could look something like this:
```prisma
model User {
  id        Int         @id @default(autoincrement())
  email     String      @unique
  name      String
  createdAt DateTime    @default(now())
  updatedAt DateTime    @updatedAt
  costData CostData[]
}

model CostData {
  id        Int      @id @default(autoincrement())
  userId   Int
  owner     User     @relation(fields: [userId], references: [id])
  timestamp DateTime @db.Timestamp(6)
  value     Int
}
```

**Reply by pantharshit00** [reply]
👋 Hello @laurent512,  did you perhaps have a chance to check @nurul3101's recommendation? Let us know if you still have any questions!
 
 Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
